examples/sparse/unignn.py

```diff
@@ -0,0 +1,276 @@
+"""`UniGNN <https://arxiv.org/abs/2105.00956>`_
+
+This script reproduces the result from the
+`official repository <https://github.com/OneForward/UniGNN>`_
+with the following command:
+
+.. code::
+
+   python train.py --data=cocitation --dataset=cora --model-name=UniGAT
+"""
+import pickle
+import os
+import contextlib
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import tqdm
+import dgl
+import dgl.data.utils
+import dgl.nn
+import dgl.function as fn
+import numpy as np
+
+@contextlib.contextmanager
+def changedir(path):
+    try:
+        cwd = os.getcwd()
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(cwd)
+
+class HyperGCNCocitationCora(dgl.data.DGLDataset):
```

Comment:
Why not make it a standard data set in DGL?

Reply:
This is to show you how a hypergraph dataset will look like.  If you guys are OK to make it a standard DGL dataset then I'll open another PR.

Reply:
+1 for moving it to DGL core